### DISCUSSION TITLE
Can I manually create ViewModels instead of Prism?

### DISCUSSION THREAD

**Discussion by achobanov** [PrismLibrary/Prism#2558] [WPF]
Essentially, I need to manage the reference to a ViewModel, before I navigate to it's view. Is this possible?

**Comment by sigmarsson** [comment]
Just inject it whenever you need those references.
i needed something similar earlier.
https://github.com/PrismLibrary/Prism/discussions/2496

**Reply by achobanov** [reply]
@sigmarsson thanks for the response, but I'm trying to do something different. I have structed my views in a tree. Meaning that I have a root view and beneath it are nodes of parents and children. I've designed it like that because part of my requirements is that I am able to represent the data for all child views within the current parent view. 

Let say that I have nodes `A`, `B` and `C`. 
1. `A` has a list of `B` nodes
2. `B` has a single `C` node

_Each node is an instance of it's view. There are multiple instances of all nodes, aka multiple views of the same type:_

At the level of node `B`  I have to render `B`-specific content and `C`-specific content. And up one level - on `A` I need `A`-specific content and a list of `B` aggregates, each showing some data from `B` and `C`. 

In order to achieve that I'm tracking every child VIewModels in a collection in their parent, up until the root ViewModel. The models are mapped from data as it comes in the data layer. However when I navigate to a new View instance, Prism creates it's own instance of a ViewModel instead of using mine directly. My current approach is to map the former from the latter, but I end up with two objects, rather than one. That causes problems, as you might imagine. 

Ideally I would be able to instantiate my own ViewModel, add it to a collection and pass that same object to Prism to render. Then due to reference equality, my data structure state will represent the actual state from the ViewModel.

**Comment by brianlagunas** [comment]
If you want to stop prism from automatically creating the ViewModel just set the AutowireViewModel attached property to false on the view.

**Comment by mxsoftware** [comment]
One possible approach is to use ViewModelLocationProvider.SetDefaultViewModelFactory
to define your own view model creation strategies.
